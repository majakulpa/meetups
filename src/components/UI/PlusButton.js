import React from 'react'
import { HiPlus } from 'react-icons/hi'

const plusButton = ({ click, text, plus }) => {
  return (
    <button
      className="block bg-purple-600 float-right hover:bg-purple-800 text-white tracking-wide flex
        capitalize py-2 px-4 rounded focus:bg-purple-800 focus:outline-none focus:shadow-outline"
      onClick={click}
    >
      {plus ? <HiPlus className="mt-1 mr-1 font-bold" /> : ''}

      <span>{text}</span>
    </button>
  )
}

export default plusButton
