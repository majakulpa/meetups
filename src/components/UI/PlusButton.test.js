import React from 'react'
import ReactDOM from 'react-dom'
import '@testing-library/jest-dom/extend-expect'
import { render, cleanup, fireEvent } from '@testing-library/react'
import PlusButton from './PlusButton'
import renderer from 'react-test-renderer'

afterEach(cleanup)

test('if renders', () => {
  const div = document.createElement('div')
  ReactDOM.render(<PlusButton />, div)
})

test('if matches snapshot', () => {
  const tree = renderer.create(<PlusButton />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('if clicking the button calls event handler once', () => {
  const mockHandler = jest.fn()

  const component = render(<PlusButton text="Add" click={mockHandler} />)

  const button = component.getByText('Add')
  fireEvent.click(button)

  expect(mockHandler.mock.calls).toHaveLength(1)
})
