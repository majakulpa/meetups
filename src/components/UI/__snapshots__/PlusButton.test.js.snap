// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if matches snapshot 1`] = `
<button
  className="block bg-purple-600 float-right hover:bg-purple-800 text-white tracking-wide flex capitalize py-2 px-4 rounded focus:bg-purple-800 focus:outline-none focus:shadow-outline"
  id="plusButton"
  onClick={undefined}
>
  
  <span
    className="text-xs sm:text-base"
  />
</button>
`;
